client/extensions/hello-dolly/state/selectors.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Internal Dependencies
+ */
+import lyrics from './lyrics';
+
+export function getCurrentLyric( reduxState ) {
+	const state = reduxState.helloDolly;
+
+	return lyrics[ state.lyricIndex ];
```

Comment:
just a quick note on this example - it's open to type errors if `reduxState.helloDolly` isn't available (`state.lyricIndex`) and range errors if the index is longer than the lyrics array. I'm a big fan of using `lodash.get()` with a default value which we know if _always_ safe to use, but if we don't want to pull in `lodash` for some reason it might be wise to guard ourselves here.

```js
export const getCurrentLyric = state => {
	// default to first lyric if unavailable
	const index = get( state, 'helloDolly.lyricIndex', 0 );

	// wrap around if given too high of an index
	return lyrics[ index % lyrics.length ];
```

you may notice that I made some decisions in this code that could be decided differently: what happens when the state fails us? here I decided that we should simply wrap back to the beginning. maybe an empty string or `null` would have also made good return values.